File: client/src/types/booking.ts
export interface Booking {
  id: string;
  booking_reference: string;
  user_id: string;
  trip_id: string;
  status: string;
  payment_flow: string;
  total_amount: number;
  paid_amount: number;
  paypal_order_id: string | null;
  authorization_id: string | null;
  authorization_expires_at: string | null;
  vault_token_id: string | null;
  invoice_id: string | null;
  invoice_url: string | null;
  created_at: string;
  updated_at: string;
  // Joined fields
  trip_name?: string;
  trip_slug?: string;
  image_gradient?: string;
  customer_name?: string;
  customer_email?: string;
}

export interface BookingCharge {
  id: string;
  booking_id: string;
  type: string;
  description: string;
  amount: number;
  paypal_capture_id: string | null;
  status: string;
  created_at: string;
}

export interface BookingDetail extends Booking {
  charges: BookingCharge[];
  base_price?: number;
  trip_deposit_amount?: number;
  duration_days?: number;
  trip_description?: string;
  trip_payment_flow?: string;
}

File: ios/trip-ios/trip-ios/Models/Booking.swift
import Foundation

struct Booking: Codable, Identifiable, Sendable {
    let id: String
    let bookingReference: String
    let userId: String
    let tripId: String
    let status: String
    let paymentFlow: String
    let totalAmount: Double
    let paidAmount: Double
    let paypalOrderId: String?
    let authorizationId: String?
    let authorizationExpiresAt: String?
    let vaultTokenId: String?
    let invoiceId: String?
    let invoiceUrl: String?
    let pickupDate: String?
    let dropoffDate: String?
    let createdAt: String
    let updatedAt: String

    // Joined fields from server
    let tripName: String?
    let tripSlug: String?
    let imageGradient: String?
    let customerName: String?
    let customerEmail: String?
}

struct BookingCharge: Codable, Identifiable, Sendable {
    let id: String
    let bookingId: String
    let type: String
    let description: String
    let amount: Double
    let paypalCaptureId: String?
    let status: String
    let createdAt: String
}

struct BookingDetail: Codable, Sendable {
    let id: String
    let bookingReference: String
    let userId: String
    let tripId: String
    let status: String
    let paymentFlow: String
    let totalAmount: Double
    let paidAmount: Double
    let paypalOrderId: String?
    let authorizationId: String?
    let authorizationExpiresAt: String?
    let vaultTokenId: String?
    let invoiceId: String?
    let invoiceUrl: String?
    let pickupDate: String?
    let dropoffDate: String?
    let createdAt: String
    let updatedAt: String
    let tripName: String?
    let tripSlug: String?
    let imageGradient: String?
    let customerName: String?
    let customerEmail: String?
    let charges: [BookingCharge]
    let basePrice: Double?
    let tripDepositAmount: Double?
    let durationDays: Int?
    let tripDescription: String?
    let tripPaymentFlow: String?
}

struct MerchantStats: Codable, Sendable {
    let activeBookings: Int
    let pendingCaptures: Int
    let openInvoices: Int
    let monthlyRevenue: Double
}
